### ISSUE TITLE
How to get the progress of writing to the stream

### ISSUE THREAD

**Issue by Lee-42** [theophilusx/ssh2-sftp-client#310]

**Comment by theophilusx** [comment]
Streams just use the pip() call, so you can chain in whatever additional
processing you want. If, for example, you want something to display the
progress, you could implement that as your own pass-through type stream
which takes one stream as input, keeps track of bytes seen, displays
some sort of update count/graph/whatever, and then passes that data out
to be consumed by the next stream in the pipe chain.

There are likely npm modules which already provide such functionality.

So basically, this is up to you on how you want to implement such a
feature. It is not part of the ssh2-sftp-client module. However, the
module will work fine with such 'plugins'. The only limitation is that
you have to use the get()/put() methods (not fastGet/fastPut as they
don't support passing in/out streams.

**Comment by Lee-42** [comment]
Thank you very much for your answer, I successfully solved the problem
